### PR TITLE
Ft gymnasium migration

### PR DESCRIPTION
**Pull request by maxspahn**
Migrating to gymnasium, addresses #192 
Have a look, @behradkhadem.
Might also solves issues mentioned in #190.

I had a nice morning in the park and realized the migration is rather straight-forward.

### CODE REVIEW

**Comment by behradkhadem** [comment]
Nice job! I myself was working on this issue too ([here](https://github.com/behradkhadem/gym_envs_urdf/commit/c798bba5bf074371c709753e21f01ffdd2b72f88)) and I encountered the same error that your version has. While running `env_checker.py` I get this error: 
```
Traceback (most recent call last):
  File "/home/behradx/anaconda3/envs/SB3/lib/python3.9/site-packages/stable_baselines3/common/env_checker.py", line 402, in check_env
    env.reset(seed=0)
  File "/home/behradx/anaconda3/envs/SB3/lib/python3.9/site-packages/gymnasium/core.py", line 462, in reset
    obs, info = self.env.reset(seed=seed, options=options)
TypeError: reset() got an unexpected keyword argument 'options'

The above exception was the direct cause of the following exception:

Traceback (most recent call last):
  File "/home/behradx/projects/gym_envs_urdf/examples/reinforcement learning/env_checker.py", line 76, in <module>
    check_env(env, warn=True)
  File "/home/behradx/anaconda3/envs/SB3/lib/python3.9/site-packages/stable_baselines3/common/env_checker.py", line 404, in check_env
    raise TypeError("The reset() method must accept a `seed` parameter") from e
TypeError: The reset() method must accept a `seed` parameter
```

In my version, I added `seed` parameter to the `reset` method but still got the same error. I couldn't pinpoint the source of the issue. I also had a problem with nested `Dict` for observation. It seems that your implementation doesn't have this problem.

PS: Stable baselines 3 for `gymnasium` could be installed via `pip install "stable_baselines3[extra]>=2.0.0a9"`.

**Comment by maxspahn** [comment]
@behradkhadem , great that you are so responsive!!! It helps a lot! I'll integrate it today. Actually, I should simply make you a contributor, that you can change it!

**Comment by behradkhadem** [comment]
> @behradkhadem , great that you are so responsive!!! It helps a lot! I'll integrate it today. Actually, I should simply make you a contributor, that you can change it!

I'd be happy to help. I tried pushing code but I didn't have the access. 
```
ERROR: Permission to maxspahn/gym_envs_urdf.git denied to behradkhadem.
fatal: Could not read from remote repository.

Please make sure you have the correct access rights
and the repository exists.
```

**Comment by maxspahn** [comment]
I sent you the invite to become a contributor. Please continue discussing merge requests and don't abuse the power ;)

File: examples/reinforcement learning/env_checker.py [hunk 1]
```diff
@@ -0,0 +1,71 @@
...
+    - Whether the environment uses a `VecEnv` or not
+    - Whether the environment uses a `VecNormalize` wrapper or not
+    - Whether the environment uses a `FlattenObservation` wrapper or not
+    - Whether the environment uses a `FrameStack` wrapper or not
+    - Whether the environment uses a `TimeLimit` wrapper or not
+    - Whether the environment uses a `Monitor` wrapper or not
+    - Whether the environment uses a `VecFrameStack` wrapper or not
+    - Whether the environment uses a `VecTransposeImage` wrapper or not
+'''
+from stable_baselines3.common.env_checker import check_env
```
Comment (maxspahn) [inline examples/reinforcement learning/env_checker.py:26]: This introduces a dependency on stable baselines. However, RL is one use case and urdfenvs should done add dependencies that are specific do a use case.

I think the whole folder should not be in here.

File: urdfenvs/urdf_common/urdf_env.py [hunk 2]
```diff
@@ -247,8 +247,15 @@ def step(self, action):
 
         if self._render:
             self.render()
-        return ob, reward, self._done, self._info
 
+        terminated = self._done
+        truncated = self._getTruncated()
+        return ob, reward, terminated, truncated, self._info
+    
+    def _getTruncated(self) -> bool:
```
Comment (maxspahn) [inline urdfenvs/urdf_common/urdf_env.py:255]: Can you respect the pep guidelines for method naming, this should be 'get_truncated'. Can you check it for the other functions as well.

File: urdfenvs/urdf_common/urdf_env.py [hunk 3]
```diff
@@ -534,11 +541,13 @@ def add_sensor(self, sensor: Sensor, robot_ids: List) -> None:
 
     def reset(
         self,
+        seed = None, 
+        options = None,
```
Comment (maxspahn) [inline urdfenvs/urdf_common/urdf_env.py:545]: Can you add the type annotation for those?

**Review by maxspahn — COMMENTED** [review]
Great @behradkhadem !!! However, I would like to remove the examples for RL, as it introduces a dependency on stable baselines in this case. Then, there are some minor comments.

File: examples/reinforcement learning/env_checker.py (hunk 1 above)
Comment (behradkhadem) [reply examples/reinforcement learning/env_checker.py:26]: But I think it's ok to have `SB3Float32ActionWrapper`, don't you think? It does not add any dependency and uses gymnasium.

File: urdfenvs/urdf_common/urdf_env.py (hunk 2 above)
Comment (behradkhadem) [reply urdfenvs/urdf_common/urdf_env.py]: Ok.

File: urdfenvs/urdf_common/urdf_env.py (hunk 3 above)
Comment (behradkhadem) [reply urdfenvs/urdf_common/urdf_env.py]: Done.

**Comment by behradkhadem** [comment]
Dear @maxspahn, everything seems fine to me. I think now we can use this package for RL purposes. I'll try to do a simple pick and place using panda robot. Just a few questions:

- We can import and use our own URDF files, can't we? I'm asking because I'm planning to design a robot for my masters project (in SolidWorks) and then [convert it to a URDF file](http://wiki.ros.org/sw_urdf_exporter/Tutorials/Export%20an%20Assembly). 
- Do we have a camera sensor? I remember seeing something about it, but I fail to find it now.
- Are there anything else that I could help developing? I suggest adding a few `.md` files for future steps and contribution conditions.

Many thanks again!

**Comment by maxspahn** [comment]
> We can import and use our own URDF files, can't we? I'm asking because I'm planning to design a robot for my masters project (in SolidWorks) and then convert it to a URDF file.

Yes, you can. However, the robot you are creating must be of one of the supported types (holonomic, diff-drive or bicycle model).

> Do we have a camera sensor? I remember seeing something related to it inside the package, but I fail to find it now.

No, there is no camera sensor so far.

> Are there anything else that I could help developing? I suggest adding a few .md files for future steps and contribution conditions.

Maybe, you could add the camera sensor :smile: . Those contributing files would also be nice to have.

**Pull request merged**